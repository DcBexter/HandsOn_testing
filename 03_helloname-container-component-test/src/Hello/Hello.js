import React from 'react';
import PropTypes from 'prop-types';

const Hello = ({ age, name, index, changeName }) => {
    return (
      <>
        <input onChange={event => changeName(event, index)}/>
        <h1>{`I'm ${name} and i am ${age} years old!`}</h1>
      </>
    );
}

Hello.propTypes = {
  name: PropTypes.string.isRequired,
  age: PropTypes.number.isRequired,
  index: PropTypes.number.isRequired,
  changeName: PropTypes.func.isRequired
};

export default Hello;
