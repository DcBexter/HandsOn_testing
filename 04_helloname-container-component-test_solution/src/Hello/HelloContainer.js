import React, { Component } from 'react';
import Hello from './Hello'

const persons = [
  {
    name: 'Anton',
    age: 23,
    id: 4711
  },
  {
    name: 'Benno',
    age: 21,
    id: 1231
  },
  {
    name: 'Steffen',
    age: 32,
    id: 2234
  },
]

class HelloContainer extends Component {

  constructor(props){
    super(props);
    this.state = {
      persons,
    };
  }

  changeNameHandler = (event, index) => {
    const updatedPersons = [...this.state.persons];
    updatedPersons[index].name = event.target.value;
    this.setState({
        persons: updatedPersons,
    });
  }

  render() {
    return (
      <>
        {this.state.persons.map((person, index) =>
          <Hello
            key={person.id}
            index={index}
            name={person.name}
            age={person.age}
            changeName={this.changeNameHandler}
          />
        )}
      </>
    );
  }
}

export default HelloContainer;
