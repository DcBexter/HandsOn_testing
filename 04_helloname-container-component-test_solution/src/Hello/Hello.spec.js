import React from "react";
import { shallow } from "enzyme";
import Hello from "./Hello";
import renderer from "react-test-renderer";

const fn = jest.fn();

describe("hello component", () => {
  it("should render without crash", () => {
    shallow(<Hello name="Markus" age={20} index={0} changeName={fn} />);
  });

  it("contains correct text", () => {
    const wrapper = shallow(
      <Hello name="Markus" age={20} index={0} changeName={fn} />
    );
    expect(wrapper.text()).toEqual("I'm Markus and i am 20 years old!");
  });

  it("change event handler should be called with correct value", () => {
    const fn = jest.fn();
    const wrapper = shallow(
      <Hello name="Markus" age={20} index={0} changeName={fn} />
    );
    const event = { target: { value: "Werner" } };
    wrapper.find("input").simulate("change", event);
    expect(fn).toBeCalledWith(event, 0);
  });

  it("hello component matches expected snapshot", () => {
    const tree = renderer
      .create(<Hello name="Markus" age={20} index={0} changeName={fn} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
