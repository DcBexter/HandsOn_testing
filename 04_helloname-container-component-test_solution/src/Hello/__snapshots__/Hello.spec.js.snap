// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hello component hello component matches expected snapshot 1`] = `
Array [
  <input
    onChange={[Function]}
  />,
  <h1>
    I'm Markus and i am 20 years old!
  </h1>,
]
`;
